src/libs/EmojiUtils.ts

```diff
@@ -477,109 +452,87 @@ const getPreferredEmojiCode = (emoji, preferredSkinTone) => {
 
-            if (!oldestEmojiTimestamp || oldestUserTimestamp < oldestEmojiTimestamp) {
-                oldestEmojiTimestamp = oldestUserTimestamp;
-            }
+        if (!oldestEmojiTimestamp || oldestUserTimestamp < oldestEmojiTimestamp) {
+            oldestEmojiTimestamp = oldestUserTimestamp;
+        }
 
-            return {
-                ...user,
-                id,
-                oldestTimestamp: oldestUserTimestamp,
-            };
-        })
-        .value();
+        usersWithTimestamps[id] = {
+            ...user,
+            id,
+            oldestTimestamp: oldestUserTimestamp,
+        };
+    });
 
     return {
         ...emoji,
         users: usersWithTimestamps,
         // Just in case two emojis have the same timestamp, also combine the timestamp with the
         // emojiName so that the order will always be the same. Without this, the order can be pretty random
         // and shift around a little bit.
-        oldestTimestamp: (oldestEmojiTimestamp || emoji.createdAt) + emojiName,
+        oldestTimestamp: (oldestEmojiTimestamp ?? emoji.createdAt) + emojiName,
     };
 };
 
 /**
  * Returns true if the accountID has reacted to the report action (with the given skin tone).
  * Uses the NEW FORMAT for "emojiReactions"
- * @param {String} accountID
- * @param {Array<Object | String | number>} usersReactions - all the users reactions
- * @param {Number} [skinTone]
- * @returns {boolean}
+ * @param usersReactions - all the users reactions
  */
-function hasAccountIDEmojiReacted(accountID, usersReactions, skinTone) {
-    if (_.isUndefined(skinTone)) {
+function hasAccountIDEmojiReacted(accountID: string, usersReactions: TimestampedUsersReactions, skinTone?: number) {
+    if (skinTone === undefined) {
         return Boolean(usersReactions[accountID]);
     }
     const userReaction = usersReactions[accountID];
-    if (!userReaction || !userReaction.skinTones || !_.size(userReaction.skinTones)) {
+    if (!userReaction?.skinTones || !Object.values(userReaction.skinTones).length) {
         return false;
     }
     return Boolean(userReaction.skinTones[skinTone]);
 }
 
 /**
  * Given an emoji reaction and current user's account ID, it returns the reusable details of the emoji reaction.
- * @param {String} emojiName
- * @param {Object} reaction
- * @param {String} currentUserAccountID
- * @returns {Object}
  */
-const getEmojiReactionDetails = (emojiName, reaction, currentUserAccountID) => {
+const getEmojiReactionDetails = (emojiName: string, reaction: UsersReactionsList, currentUserAccountID: string) => {
     const {users, oldestTimestamp} = enrichEmojiReactionWithTimestamps(reaction, emojiName);
 
     const emoji = findEmojiByName(emojiName);
     const emojiCodes = getUniqueEmojiCodes(emoji, users);
-    const reactionCount = lodashSum(_.map(users, (user) => _.size(user.skinTones)));
+    const reactionCount = Object.values(users)
+        .map((user) => Object.values(user.skinTones).length)
```

Comment:
```suggestion
        .map((user) => Object.values(user.skinTones ?? {}).length)
```

Reply:
Crash doesn't happen anymore after this change

Reply:
Done! I've also added this in a few other places, to make sure that potential `undefined` are handled.